internal/apis/certmanager/v1alpha2/types_certificate.go

```diff
@@ -275,6 +275,15 @@ type X509Subject struct {
 	// Serial number to be used on the Certificate.
 	// +optional
 	SerialNumber string `json:"serialNumber,omitempty"`
+	// Extra names to be used on the Certificate.
+	ExtraNames []ExtraName `json:"extraNames,omitempty"`
+}
+
+type ExtraName struct {
+	// The type. This should be an asn1.ObjectIdentifier string, e.g., "2.5.4.42"
+	Type string `json:"type,omitempty"`
+	// The value. This should be an IA5String
+	Value string `json:"value,omitempty"`
```

Comment:
The 'type' and 'value' fields are presumably not optional, so we shouldn't have 'omitempty' here 😊

Also there should be validation on both of these fields - eg the type can be checked to ensure it's a valid identifier, and then I'm uncertain precisely what validation we can perform on the value, but at the very least a maxLength check should be in place.

Reply:
Hm, I do agree that the Type shouldn't be empty. However, I don't think anything prevents one from having a type with an empty value. I'm not sure what the use-case would be, but technically, null values are allowed for values because null is a valid IA5 character.

Reply:
On second thought... if it is empty, i think it should show as empty, so yes, deleting both omitEmpty annotations is probably correct.